test/xref2/github_issue_857.t/a.mli

```diff
@@ -0,0 +1,7 @@
+module type A = sig
+
+  (** {1:first First inner section} *)
+
+end
+
+(** {1:first First outer section} *)
```

Comment:
To best reproduce the issue, you need a reference to the outer section and to check that the generated link is correct.
Also, `A` doesn't need to be a module type.

Reply:
In fact, the PR does not rewrite references: it only fixes the label disambiguation, which was too aggressive in the HTML case.

Label disambiguation cannot be left to the user: two titles with the same name (which is valid) would generate the same label, which would need disambiguation. On the contrary, references should not be rewritten (I think): for instance, in:
```ocaml
(** {1 A title} *)
(** {1 A title} *)

(** reference to {!a-title} *)
(** reference to {!a-title_2} *)
```
the first reference points to the first title, the second to the second, and none of them should be rewritten.

Reply:
You are right. No need to add references as we know what they'll do.

Disambiguated labels are used in the TOC however, which would make sense to grep.